Triangulation_3/include/CGAL/IO/Triangulation_raw_iostream_3.h

```diff
@@ -0,0 +1,462 @@
+bool write_header(std::ostream& os, const Triangulation_3<GT, Tds_, Lds_>& tr)
+{
+  // Writes:
+  // - the dimension
+  if(IO::is_ascii(os))
+  {
+    os << tr.dimension() << "\n";
+  }
+  else
+  {
+    write(os, tr.dimension());
+  }
+  return (bool)os;
+}
+
+template < class GT, class Tds_, class Lds_ >
+bool read_header(std::istream& is, Triangulation_3<GT, Tds_, Lds_>& tr)
+{
+  // Reads:
+  // - the dimension
+  int dimension;
+  if(IO::is_ascii(is))
+  {
+    is >> dimension;
+  }
+  else
+  {
+    read(is, dimension);
+  }
+  if(dimension > 3 || dimension < -2)
+    return false;
+  tr.tds().set_dimension(dimension);
+  return (bool)is;
+}
+
+template < class GT, class Tds_, class Lds_ >
+bool write_vertices(std::ostream& os, const Triangulation_3<GT, Tds_, Lds_>& tr)
+{
+  // Writes:
+  // - the number of finite vertices
+  // - the non combinatorial information on vertices (point, etc)
+  typedef Triangulation_3<GT, Tds_>               Triangulation;
+  typedef typename Triangulation::size_type       size_type;
+  typedef typename Triangulation::Vertex_iterator Vertex_iterator;
+
+  size_type n = tr.number_of_vertices();
+
+  if(IO::is_ascii(os))
+  {
+    os << n << '\n';
+  }
+  else
+  {
+    write(os, n);
+  }
+
+  if (n == 0)
+    return false;
+
+  for(Vertex_iterator it = ++tr.vertices_begin(), end = tr.vertices_end(); it != end; ++it)
+  {
+    os << *it;
+    if(IO::is_ascii(os))
+      os << '\n';
+  }
+
+  return (bool)os;
+}
+
+template < class GT, class Tds_, class Lds_,
+           typename Tr_src = Triangulation_3<GT, Tds_, Lds_>,
+           typename ConvertVertex = Nullptr_t >
+struct read_vertices_functor
+{
+  bool operator()(std::istream& is, Triangulation_3<GT, Tds_, Lds_>& tr,
+                  std::vector< typename Triangulation_3<GT, Tds_>::Vertex_handle >& vertices_handles,
+                  typename Triangulation_3<GT, Tds_>::size_type n,
+                  ConvertVertex convert_vertex = ConvertVertex())
+  {
+    // Reads:
+    // - the non combinatorial information on vertices (point, etc)
+    // The vertices are read with the type Tr_src::Vertex
+    // and converted to type Triangulation_3<GT, Tds_, Lds_>::Vertex
+    // using the ConvertVertex convertor.
+    typedef typename Tr_src::Vertex Vertex1;
+
+    for(std::size_t i=1; i <= n; i++)
+    {
+      Vertex1 v;
+      if(!(is >> v))
+        return false;
+      vertices_handles[i] = tr.tds().create_vertex();
+      convert_vertex(v, *vertices_handles[i]);
+    }
+
+    return (bool)is;
+  }
+};
+
+template < class GT, class Tds_, class Lds_, class Tr_src >
+struct read_vertices_functor < GT, Tds_, Lds_, Tr_src, Nullptr_t >
+{
+  bool operator()(std::istream& is, Triangulation_3<GT, Tds_, Lds_>& tr,
+                  std::vector< typename Triangulation_3<GT, Tds_>::Vertex_handle >& vertices_handles,
+                  typename Triangulation_3<GT, Tds_>::size_type n,
+                  Nullptr_t _ = Nullptr_t())
```

Comment:
This variable has a strange name and is not used leading to [warnings](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.0-Ic-117) in the testsuite.
I am wondering if it is also responsible for the errors for example [here](https://cgal.geometryfactory.com/CGAL/testsuite/#Hash_map).

Reply:
The variable [`_` will have a very special role in C++26 (accepted paper P2169r4)](https://wg21.link/p2169r4): in particular it will have an implicit `[[maybe_unused]]`  attribute.

But for a function parameter the easiest is to remove the parameter name: `Nullptr_t  = Nullptr_t())` to remove the warning.